src/ui/handlers/abstract-ui-handler.ts

```diff
@@ -18,25 +36,133 @@ export abstract class UiHandler {
+  }
+
+  public get ready(): boolean {
+    return this._ready;
+  }
+
+  /**
+   * Prepares the handler for display.
+   * Should not be overridden. Calls {@linkcode setup}, which subclasses should override.
+   */
+  public initialize(): void {
+    if (this._ready) {
+      console.warn("Attempting to initialize an already ready handler. Aborting.");
+      return;
+    }
+
+    this.setup();
+    this._ready = true;
+  }
+
+  /**
+   * Destroys the handler and makes it no longer usable.
+   * Should not be overridden. Calls {@linkcode tearDown}, which subclasses should override.
+   */
+  public destroy(): void {
+    if (!this.ready) {
+      console.warn("Attempting to destroy a non initialized handler. Aborting.");
+      return;
+    }
+
+    // If the handler is active, stop it first.
+    if (this.active) {
+      this.stop();
+    }
+
+    this.tearDown();
+    this._ready = false;
+  }
+
+  /**
+   * Displays the handler.
+   * Should not be overridden. Calls {@linkcode show}, which subclasses should override.
+   *
+   * @param args the arguments needed by this handler for display.
+   * @returns `true` if the handler was setup successfully, `false` otherwise.
+   */
+  public start(...args: unknown[]): boolean {
+    if (!this.ready) {
+      console.warn("Attempting to start a non ready handler. Aborting.");
+      return false;
+    }
 
-  show(..._args: unknown[]): boolean {
-    this.active = true;
+    const success = this.show(...args);
+    this._active = true;
+    return success;
+  }
 
-    return true;
+  /**
+   * Ends display of the handler.
+   * Should not be overridden. Calls {@linkcode clear}, which subclasses should override.
+   */
+  public stop(): void {
+    if (!this.active) {
+      console.warn("Attempting to stop a non active handler. Aborting.");
+      return;
+    }
+
+    this.clear();
+    this._active = false;
   }
 
-  abstract processInput(button: Button): boolean;
+  /**
+   * Create the basic building blocks of the ui handlers that will stay as long as it's meant to be used.
+   * In general, we add basic containers or elements to `this.getUi()` here.
+   *
+   * Elements (containers, listeners, ...) created here should be destroyed in {@linkcode tearDown}.
+   */
+  protected abstract setup(): void;
 
-  getUi() {
+  /**
+   * Called when the handler will never be used again.
+   *
+   * Subclasses should destroy any element initialized in {@linkcode setup} here.
+   * For most handlers, calling `mainContainer.destroy()` and removing any listeners
+   * created by the handler should be enough.
+   */
+  protected abstract tearDown(): void;
+
+  /**
+   * Called when the mode corresponding this handler is set, and it should
+   * prepare the handler for display, based on the arguments provided.
+   *
+   * Elements (containers, gameObjects, listeners, ...) created here should be destroyed in {@linkcode clear}.
+   *
+   * @param _args the arguments needed by this handler for display.
```

Comment:
NIT:  Recently learned that the dash `-` is supposed to be put manually.
```suggestion
   * @param _args - The arguments needed by this handler for display.
```